src/main/java/org/openlmis/requisition/domain/JasperTemplateParameterDependency.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+package org.openlmis.requisition.domain;
+
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Table(name = "jasper_template_parameter_dependencies")
+@NoArgsConstructor
+public class JasperTemplateParameterDependency extends BaseEntity {
+  @ManyToOne(cascade = CascadeType.REFRESH)
+  @JoinColumn(name = "parameterId", nullable = false)
+  @Getter
+  @Setter
+  private JasperTemplateParameter parameter;
+
+  @Column(columnDefinition = TEXT_COLUMN_DEFINITION, nullable = false)
+  @Getter
+  @Setter
+  private String dependency;
+
+  @Column(columnDefinition = TEXT_COLUMN_DEFINITION, nullable = false)
+  @Getter
+  @Setter
+  private String placeholder;
```

Comment:
Basically we want to be able to change selectExpression for parameter, depending on other parameters' values (those would be query parameters for url).